modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/components/SearchField.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {ClayButtonWithIcon} from '@clayui/button';
+import {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import React, {useEffect, useState} from 'react';
+
+const ITEM_TYPES_SYMBOL = {
+	article: 'document-text',
+	folder: 'folder',
+};
+
+const SEARCH_DELTA = 2;
+
+export default function SearchField({handleSearchChange, items}) {
+	const initialState = {
```

Comment:
Hi @ambrinchaudhary :)
Do you think we could change some of the names to make them more self-explanatory? 
What do you think about `initialSearchInfo` for this?

Reply:
Yeah, I think it makes sense, I'll change it!